src/unity.c

```diff
@@ -445,7 +445,7 @@ void UnityPrintFloat(const UNITY_DOUBLE input_number)
 
         /* build up buffer in reverse order */
         digits = 0;
-        while ((n != 0) || (digits < (decimals + 1)))
```

Comment:
Do you need this check at all ? As digits is always zero and decimals can't be less than zero?

Reply:
digits is incremented in the loop.

Reply:
Argh my bad for missing that! Just smile and wave boys :D